### PR TITLE
Use embedded PostgreSQL and MySQL for testing

### LINKED ISSUE #112: Use embedded PostgreSQL and MySQL for testing
Currently the tests require to setup a database for running them. We can use Java projects that allow to programmatically start/stop databases for testing purposes instead.

### PR DESCRIPTION
**Pull request by vietj**
Fixes #112

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -14,12 +14,28 @@ Please see the in source asciidoc documentation or the main documentation on the
...
 
-== Requirements for running test suite
+== Running test suite
+
+The test suite embeds PostgreSQL and MySQL with
+
+* ru.yandex.qatools.embed:postgresql-embedded
+* com.wix:wix-embedded-mysql
+
+You can run provide the servers instead by specifying the `host` property
```
Comment (tsegismont) [inline README.adoc]: _can run provide_ ?

Comment (tsegismont) [inline README.adoc]: Change to: _You can also run your own server instead by providing the `host` property_

File: README.adoc [hunk 2]
```diff
@@ -14,12 +14,28 @@ Please see the in source asciidoc documentation or the main documentation on the
...
+
+----
+> mvn test -Dhost=localhost
+----
 
-* A working PostgreSQL or MySQL server
 * For testing PostgreSQL: A `testdb` database on a local PostgreSQL install and a user called `vertx` with password `password`
 * For testing MySQL: A `testdb` database on a local MySQL install and a user called `vertx` with password `password`
 
+If needed test ports can be overriden
```
Comment (Narigo) [inline README.adoc]: Test ports can be changed, if necessary.

File: src/test/java/io/vertx/ext/asyncsql/SQLTestBase.java [hunk 3]
```diff
@@ -46,6 +46,14 @@
 
 public abstract class SQLTestBase extends AbstractTestBase {
 
+  static final String POSTGRESQL_HOST = System.getProperty("postgresql.host", "localhost").trim();
```
Comment (Narigo) [inline src/test/java/io/vertx/ext/asyncsql/SQLTestBase.java]: Should we have a `POSTGRESQL_SSL_HOST` as well?

File: src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java [hunk 4]
```diff
@@ -74,7 +91,7 @@ public void testWrongSslConfiguration(TestContext context) {
     Async async = context.async();
     client = createClient(vertx,
       new JsonObject()
-        .put("host", System.getProperty("db.host", "localhost"))
+        .put("host", SQLTestBase.POSTGRESQL_HOST)
         .put("port", Integer.parseInt(System.getProperty("dbssl.port", "54321")))
```
Comment (Narigo) [inline src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java]: Isn't `SQLTextBase.POSTGRESQL_SSL_PORT` missing here?

File: src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java [hunk 5]
```diff
@@ -86,12 +103,13 @@ public void testWrongSslConfiguration(TestContext context) {
...
 
+  /*
   @Test
   public void testNoSslConfiguration(TestContext context) {
     Async async = context.async();
     client = createClient(vertx,
       new JsonObject()
-        .put("host", System.getProperty("db.host", "localhost"))
+        .put("host", SQLTestBase.DB_HOST)
         .put("port", Integer.parseInt(System.getProperty("dbssl.port", "54321")))
```
Comment (Narigo) [inline src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java]: Isn't `SQLTextBase.POSTGRESQL_SSL_PORT` missing here?

File: src/test/java/io/vertx/ext/asyncsql/SQLTestBase.java (hunk 3 above)
Comment (vietj) [reply src/test/java/io/vertx/ext/asyncsql/SQLTestBase.java]: I think we can introduce it later if we need it.

File: src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java (hunk 5 above)
Comment (vietj) [reply src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java]: I don't think so as the port wasn't used before, that's how the test works

**Comment by vietj** [comment]
Docker is still used for now with Travis.

one thing I've noted is that embedded MySQL does not work in Cloudbees CI (because of missing dynamically linked io library). So I had to setup an external account to www.freemysqlhosting.net for testing.

**Pull request merged**